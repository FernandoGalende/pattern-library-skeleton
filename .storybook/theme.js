import { create } from '@storybook/theming/create';

const theme = create({
  base: 'light',
  //brandTitle: 'Pattern library skeleton',
  brandTitle: '#zapatillasFromMarsðŸ‘ŸðŸš€',
  brandUrl: 'https://github.com/klaufel/pattern-library-skeleton',
  //brandImage: 'your-image-url',
});

export default theme;
