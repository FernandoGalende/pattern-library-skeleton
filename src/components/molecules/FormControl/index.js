import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Label from '../../atoms/Label';
import Input from '../../atoms/Input';

const Wrapper = styled.div`
  padding-bottom: ${props => props.theme.spacing.md};
`;

const FormControl = ({ id, label, placeholder, status, disabled, defaultValue }) => {
  return (
    <Wrapper>
      {label && <Label htmlFor={id}>{label}</Label>}
      <Input
        id={id}
        defaultValue={defaultValue}
        status={status}
        placeholder={placeholder}
        disabled={disabled}
      />
    </Wrapper>
  );
};

FormControl.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  /** This Boolean attribute prevents the user from interacting with the button */
  disabled: PropTypes.bool,
};

export default FormControl;
