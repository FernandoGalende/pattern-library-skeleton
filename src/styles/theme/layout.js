const layout = {
  grid: {
    width: '1624px',
    columns: 12,
    gutter: {
      xs: '8px',
      sm: '16px',
      md: '24px',
      lg: '24px',
      xl: '24px'
    },
    margin: {
      xs: '16px',
      sm: '24px',
      md: '32px',
      lg: '32px',
      xl: '32px'
    }
  },
  breakpoints: {
    xs: '576px', // xs: The number of columns to span on small devices (≥576px)
    sm: '768px', // sm: The number of columns to span on medium devices (≥768px)
    md: '992px', // md: The number of columns to span on large devices (≥992px)
    lg: '1200px', // lg: The number of columns to span on extra large devices (≥1200px)
    xl: '1560px' // xl: The number of columns to span on extra large devices (≥1560px)
  }
}

export default layout
