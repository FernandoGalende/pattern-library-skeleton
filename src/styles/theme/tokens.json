{
  "colors": {
    "alertWarning": "#ffa90d",
    "alertError": "#e5363a",
    "alertSuccess": "#109c42",
    "neutral00": "#ffffff",
    "neutral05": "#f2f2f2",
    "neutral20": "#cccccc",
    "neutral40": "#999999",
    "neutral80": "#333333",
    "primaryDark": "#1d698f",
    "primaryMain": "#b8255a",
    "primaryLight": "#56a4c9"
  },
  "typography": {
    "headingBig": {
      "fontFamily": "Open Sans",
      "fontSize": "28px",
      "lineHeight": "32px",
      "fontWeight": 700
    },
    "headingNormal": {
      "fontFamily": "Open Sans",
      "fontSize": "24px",
      "lineHeight": "28px",
      "fontWeight": 700
    },
    "headingSmall": {
      "fontFamily": "Open Sans",
      "fontSize": "18px",
      "lineHeight": "21px",
      "fontWeight": 700
    },
    "bodyNormal": {
      "fontFamily": "Open Sans",
      "fontSize": "14px",
      "lineHeight": "16px",
      "fontWeight": 400
    }
  },
  "spacing": {
    "xxl": "48px",
    "xl": "40px",
    "lg": "32px",
    "md": "24px",
    "sm": "16px",
    "xs": "8px",
    "xxs": "4px"
  }
}