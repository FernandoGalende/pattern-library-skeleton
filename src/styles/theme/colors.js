import tokens from './tokens';

const colors = {
  primary: {
    light: tokens.colors.primaryLight,
    main: tokens.colors.primaryMain,
    dark: tokens.colors.primaryDark,
  },
  neutral: {
    percent00: tokens.colors.neutral00,
    percent05: tokens.colors.neutral05,
    percent20: tokens.colors.neutral20,
    percent40: tokens.colors.neutral40,
    percent80: tokens.colors.neutral80,
  },
  alert: {
    success: tokens.colors.alertSuccess,
    warning: tokens.colors.alertWarning,
    error: tokens.colors.alertError,
  },
};

export default colors;
