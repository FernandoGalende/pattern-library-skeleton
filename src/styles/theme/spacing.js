import tokens from './tokens';

const spacing = {
  xxs: tokens.spacing.xxs,
  xs: tokens.spacing.xs,
  sm: tokens.spacing.sm,
  md: tokens.spacing.md,
  lg: tokens.spacing.lg,
  xl: tokens.spacing.xl,
  xxl: tokens.spacing.xxl,
};

export default spacing;
