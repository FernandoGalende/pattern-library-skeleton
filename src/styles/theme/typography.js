import tokens from './tokens';

const typography = {
  global: {
    family: "'Open Sans'",
  },
  heading: {
    big: {
      fontSize: tokens.typography.headingBig.fontSize,
      lineHeight: tokens.typography.headingBig.lineHeight,
      fontWeight: tokens.typography.headingBig.fontWeight,
    },
    normal: {
      fontSize: tokens.typography.headingNormal.fontSize,
      lineHeight: tokens.typography.headingNormal.lineHeight,
      fontWeight: tokens.typography.headingNormal.fontWeight,
    },
    small: {
      fontSize: tokens.typography.headingSmall.fontSize,
      lineHeight: tokens.typography.headingSmall.lineHeight,
      fontWeight: tokens.typography.headingSmall.fontWeight,
    },
  },
  body: {
    normal: {
      fontSize: tokens.typography.bodyNormal.fontSize,
      lineHeight: tokens.typography.bodyNormal.lineHeight,
    },
  },
};

export default typography;
