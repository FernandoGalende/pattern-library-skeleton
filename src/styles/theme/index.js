import colors from './colors'
import typography from './typography'
import spacing from './spacing'
import layout from './layout'
import shadows from './shadows'

const theme = {
  colors,
  typography,
  spacing,
  layout,
  shadows
}

export default theme
