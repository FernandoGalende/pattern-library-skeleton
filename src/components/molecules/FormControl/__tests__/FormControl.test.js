import React from 'react'
import 'jest-styled-components'
import {render} from '@testing-library/react'
import {ThemeProvider} from 'styled-components'
import theme from '../../../../styles/theme'
import FormControl from '..'

describe('FormControl', () => {
  test('it works default', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <FormControl label="Default label" />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with status success', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <FormControl label="Default label" status="success" />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it works with status error', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <FormControl label="Default label" status="error" />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
