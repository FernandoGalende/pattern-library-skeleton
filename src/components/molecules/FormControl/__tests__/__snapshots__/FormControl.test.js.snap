// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControl it works default 1`] = `
.c1 {
  display: block;
  font-size: 14px;
  line-height: 16px;
  padding-bottom: 4px;
}

.c2 {
  background-clip: padding-box;
  background-color: #ffffff;
  background-image: none;
  border-radius: 4px;
  border: 1px solid #cccccc;
  color: #333333;
  display: block;
  font-family: inherit;
  font-size: 14px;
  line-height: 16px;
  font-weight: 400;
  height: 48px;
  outline: none;
  padding: 16px 16px;
  text-align: left;
  -webkit-transition: border 0.3s,background-color 0.3s,color 0.3s,box-shadow 0.3s;
  transition: border 0.3s,background-color 0.3s,color 0.3s,box-shadow 0.3s;
  width: 100%;
}

.c2,
.c2:hover,
.c2:focus,
.c2:active {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-shadow: none;
  outline: none;
}

.c2:hover {
  border-color: #333333;
}

.c2:focus {
  border-color: #56a4c9;
  box-shadow: 0 0 5px 0 #56a4c9;
}

.c2::-webkit-input-placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2::-moz-placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2:-ms-input-placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2::placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2:disabled,
.c2[readonly] {
  background-image: none;
  background-color: #f2f2f2;
  border-color: #cccccc;
  cursor: not-allowed;
  opacity: 1;
}

.c2:disabled,
.c2[readonly],
.c2:disabled::-webkit-input-placeholder,
.c2[readonly]::-webkit-input-placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled::-moz-placeholder,
.c2[readonly]::-moz-placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled:-ms-input-placeholder,
.c2[readonly]:-ms-input-placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled::placeholder,
.c2[readonly]::placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled:hover,
.c2[readonly]:hover {
  border-color: #cccccc;
  box-shadow: none;
}

.c2:focus {
  font-weight: 400;
}

.c2:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c2:focus::-moz-placeholder {
  opacity: 1;
}

.c2:focus:-ms-input-placeholder {
  opacity: 1;
}

.c2:focus::placeholder {
  opacity: 1;
}

.c0 {
  padding-bottom: 24px;
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    Default label
  </label>
  <input
    class="c2"
    value=""
  />
</div>
`;

exports[`FormControl it works with status error 1`] = `
.c1 {
  display: block;
  font-size: 14px;
  line-height: 16px;
  padding-bottom: 4px;
}

.c2 {
  background-clip: padding-box;
  background-color: #ffffff;
  background-image: none;
  border-radius: 4px;
  border: 1px solid #cccccc;
  color: #333333;
  display: block;
  font-family: inherit;
  font-size: 14px;
  line-height: 16px;
  font-weight: 400;
  height: 48px;
  outline: none;
  padding: 16px 16px;
  text-align: left;
  -webkit-transition: border 0.3s,background-color 0.3s,color 0.3s,box-shadow 0.3s;
  transition: border 0.3s,background-color 0.3s,color 0.3s,box-shadow 0.3s;
  width: 100%;
  background-position: top 12px right 12px;
  background-size: 24px 24px;
  background-repeat: no-repeat;
  background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgZW5hYmxlLWJhY2tncm91bmQ9Im5ldyAwIDAgMjQgMjQiPjxwYXRoIGZpbGw9IiNBOTAwMTYiIGQ9Ik0yMSAyMmgtMThjLS4zIDAtLjctLjItLjktLjUtLjItLjMtLjItLjcgMC0xbDktMThjLjMtLjcgMS40LS43IDEuOCAwbDkgMThjLjIuMy4xLjcgMCAxLS4yLjMtLjYuNS0uOS41em0tMTYuNC0yaDE0LjhsLTcuNC0xNC44LTcuNCAxNC44ek0xMSAxMGgydjZoLTJ6Ii8+PGNpcmNsZSBmaWxsPSIjQTkwMDE2IiBjeD0iMTIiIGN5PSIxOCIgcj0iMS4zIi8+PC9zdmc+');
}

.c2,
.c2:hover,
.c2:focus,
.c2:active {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-shadow: none;
  outline: none;
}

.c2:hover {
  border-color: #333333;
}

.c2:focus {
  border-color: #56a4c9;
  box-shadow: 0 0 5px 0 #56a4c9;
}

.c2::-webkit-input-placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2::-moz-placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2:-ms-input-placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2::placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2:disabled,
.c2[readonly] {
  background-image: none;
  background-color: #f2f2f2;
  border-color: #cccccc;
  cursor: not-allowed;
  opacity: 1;
}

.c2:disabled,
.c2[readonly],
.c2:disabled::-webkit-input-placeholder,
.c2[readonly]::-webkit-input-placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled::-moz-placeholder,
.c2[readonly]::-moz-placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled:-ms-input-placeholder,
.c2[readonly]:-ms-input-placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled::placeholder,
.c2[readonly]::placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled:hover,
.c2[readonly]:hover {
  border-color: #cccccc;
  box-shadow: none;
}

.c2:focus {
  font-weight: 400;
}

.c2:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c2:focus::-moz-placeholder {
  opacity: 1;
}

.c2:focus:-ms-input-placeholder {
  opacity: 1;
}

.c2:focus::placeholder {
  opacity: 1;
}

.c2,
.c2:hover {
  border-color: #e5363a;
}

.c0 {
  padding-bottom: 24px;
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    Default label
  </label>
  <input
    class="c2"
    value=""
  />
</div>
`;

exports[`FormControl it works with status success 1`] = `
.c1 {
  display: block;
  font-size: 14px;
  line-height: 16px;
  padding-bottom: 4px;
}

.c2 {
  background-clip: padding-box;
  background-color: #ffffff;
  background-image: none;
  border-radius: 4px;
  border: 1px solid #cccccc;
  color: #333333;
  display: block;
  font-family: inherit;
  font-size: 14px;
  line-height: 16px;
  font-weight: 400;
  height: 48px;
  outline: none;
  padding: 16px 16px;
  text-align: left;
  -webkit-transition: border 0.3s,background-color 0.3s,color 0.3s,box-shadow 0.3s;
  transition: border 0.3s,background-color 0.3s,color 0.3s,box-shadow 0.3s;
  width: 100%;
  background-position: top 12px right 12px;
  background-size: 24px 24px;
  background-repeat: no-repeat;
  background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgZW5hYmxlLWJhY2tncm91bmQ9Im5ldyAwIDAgMjQgMjQiPjxwYXRoIGZpbGw9IiMxMDdCMDAiIGQ9Ik04LjggMTUuOWwtNC4yLTQuMi0xLjQgMS40IDUuNiA1LjYgMTItMTItMS40LTEuNC0xMC42IDEwLjZ6Ii8+PC9zdmc+');
}

.c2,
.c2:hover,
.c2:focus,
.c2:active {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-shadow: none;
  outline: none;
}

.c2:hover {
  border-color: #333333;
}

.c2:focus {
  border-color: #56a4c9;
  box-shadow: 0 0 5px 0 #56a4c9;
}

.c2::-webkit-input-placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2::-moz-placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2:-ms-input-placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2::placeholder {
  color: #999999;
  opacity: 1;
  -webkit-transition: color 0.3s;
  transition: color 0.3s;
}

.c2:disabled,
.c2[readonly] {
  background-image: none;
  background-color: #f2f2f2;
  border-color: #cccccc;
  cursor: not-allowed;
  opacity: 1;
}

.c2:disabled,
.c2[readonly],
.c2:disabled::-webkit-input-placeholder,
.c2[readonly]::-webkit-input-placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled::-moz-placeholder,
.c2[readonly]::-moz-placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled:-ms-input-placeholder,
.c2[readonly]:-ms-input-placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled::placeholder,
.c2[readonly]::placeholder {
  color: #999999;
}

.c2:disabled,
.c2[readonly],
.c2:disabled:hover,
.c2[readonly]:hover {
  border-color: #cccccc;
  box-shadow: none;
}

.c2:focus {
  font-weight: 400;
}

.c2:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c2:focus::-moz-placeholder {
  opacity: 1;
}

.c2:focus:-ms-input-placeholder {
  opacity: 1;
}

.c2:focus::placeholder {
  opacity: 1;
}

.c2,
.c2:hover {
  border-color: #109c42;
}

.c0 {
  padding-bottom: 24px;
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    Default label
  </label>
  <input
    class="c2"
    value=""
  />
</div>
`;
